iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/typed_unique_id.inl

```diff
@@ -11,21 +11,27 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-#ifndef IOX_UTILS_CXX_TYPED_UNIQUE_ID_INL
-#define IOX_UTILS_CXX_TYPED_UNIQUE_ID_INL
+#ifndef IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_INL
+#define IOX_POSH_POPO_BUILDING_BLOCKS_TYPED_UNIQUE_ID_INL
 
 namespace iox
 {
-namespace cxx
+namespace popo
 {
 template <typename T>
 std::atomic<uint64_t> TypedUniqueId<T>::globalIDCounter{0u};
 
 template <typename T>
 inline TypedUniqueId<T>::TypedUniqueId() noexcept
-    : ThisType(newtype::internal::ProtectedConstructor, globalIDCounter.fetch_add(1u, std::memory_order_relaxed))
+    : ThisType(cxx::newtype::internal::ProtectedConstructor,
+               (static_cast<uint64_t>(internal::getUniqueRouDiId()) << 48)
+                   + ((globalIDCounter.fetch_add(1u, std::memory_order_relaxed) << 16) >> 16))
```

Comment:
Got rid of the magic numbers! Shifts are needed since the first 16-bit of this 64-bit integer is used for the RouDi id and the remaining 48-bits are used for the unique id itself.